python/lsst/cp/pipe/utils.py

```diff
@@ -58,3 +73,217 @@ def getTargetList(parsedCmd, **kwargs):
+    missing
+
+    checkTrim : `bool`
+        Check to ensure the isrTask's assembly subtask is trimming the images.
+    This is a separate config as it is very ugly to do this within the
+    normal configuration lists as it is an option of a sub task.
+
+    Raises
+    ------
+    RuntimeError
+        Raised if ``mandatory`` config parameters are False,
+        or if ``forbidden`` parameters are True.
+
+    TypeError
+        Raised if parameter ``isrTask`` is an invalid type.
+
+    Notes
+    -----
+    Logs warnings using an isrValidation logger for desirable/undesirable
+    options that are of the wrong polarity or if keys are missing.
+    """
+    if not isinstance(isrTask, ipIsr.IsrTask):
+        raise TypeError(f'Must supply an instance of ipIsr.IsrTask not {type(isrTask)}')
+
+    configDict = isrTask.config.toDict()
+
+    if logName and isinstance(logName, str):
+        log = lsstLog.getLogger(logName)
+    else:
+        log = lsstLog.getLogger("isrValidation")
+
+    if mandatory:
+        for configParam in mandatory:
+            if configParam not in configDict:
+                raise RuntimeError(f"Mandatory parameter {configParam} not found in the isr configuration.")
+            if configDict[configParam] is False:
+                raise RuntimeError(f"Must set config.isr.{configParam} to True for this task.")
+
+    if forbidden:
+        for configParam in forbidden:
+            if configParam not in configDict:
+                log.warn(f"Failed to find key {configParam} in the isr config. This is set to be " +
+                         "forbidden, so you should ensure that the equivalent for your obs_package to False.")
+                continue
+            if configDict[configParam] is True:
+                raise RuntimeError(f"Must set config.isr.{configParam} to False for this task.")
+
+    if desirable:
+        for configParam in desirable:
+            if configParam not in configDict:
+                log.info(f"Failed to find key {configParam} in the isr config. You probably want" +
+                         " to set the equivalent for your obs_package to True.")
+                continue
+            if configDict[configParam] is False:
+                log.warn(f"Found config.isr.{configParam} set to False for this task." +
+                         " It is probably desirable to have this set to True")
+    if undesirable:
+        for configParam in undesirable:
+            if configParam not in configDict:
+                log.info(f"Failed to find key {configParam} in the isr config. You probably want" +
+                         " to set the equivalent for your obs_package to False.")
+                continue
+            if configDict[configParam] is True:
+                log.warn(f"Found config.isr.{configParam} set to True for this task." +
+                         " It is probably desirable to have this set to False")
```

Comment:
I went with undesirable because it's a word, and the opposite of desirable, whereas (un)recommended isn't. It is indeed supposed to be a finger wagging, hence only being at info/warn level.

Reply:
Maybe change "It is probably desirable..." to "The cp_pipe Config recommends setting this to False."?

Reply:
Done.